includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -560,6 +577,29 @@ public function get_product() {
 		return false;
 	}
 
+	/**
+	 * Returns the login redirect URL.
+	 *
+	 * @param string $redirect Default redirect URL.
+	 * @return string Redirect URL.
+	 */
+	public function get_login_redirect_url( $redirect ) {
+		$url = esc_url_raw( wp_unslash( $_COOKIE['wcpay_payment_request_redirect_url'] ?? '' ) );
```

Comment:
is this the way we do it? I was thinking maybe the standard way was to pass the URL via query string. like we do when an unauthenticated user tries to go to a page in the admin area: 
`http://domain.tld/wp-login.php?redirect_to=http%3A%2F%2Fwcpay.jurassic.tube%2Fwp-admin%2Fadmin.php%3Fpage%3Dwc-admin&reauth=1`

I guess since we are using `woocommerce_login_redirect` filter this is how we do it.

Reply:
To redirect after login/registration within WooCommerce from the "my account" page, the only way I found was through the filters `woocommerce_login_redirect` and `woocommerce_registration_redirect`.